shared/actions/tracker.js

```diff
@@ -452,6 +452,25 @@ function _serverCallMap(
     }
   }
 
+  // We queue up responses and handle them one at a time
+  let _idleResponseQueue = []
+
+  const addToIdleResponseQueue = (f: () => void) => {
+    _idleResponseQueue.push(f)
+    requestIdle(onRequestIdleQueueHandler)
+  }
+
+  const onRequestIdleQueueHandler = () => {
+    if (!_idleResponseQueue.length) {
+      return
+    }
+    const toHandle = _idleResponseQueue.pop()
```

Comment:
:sheep:

You can check the deadline, and if we're still in the idle time we can do this again before requesting the next idle.

i.e.:

```js
const onRequestIdleQueueHandler = (deadline) => {
  while (idleResponseQueue.length && deadline.timeRemaining() > 10 /* random number: 10ms */) { 
    const toHandle = _idleResponseQueue.pop()
    toHandle()
  }

  idleResponseQueue.length && requestIdle(onRequestIdleQueueHandler)
}
```